### PR TITLE
remove iter gather

### PR DESCRIPTION
**Pull request by liqiangxl**
`IterType::Gather` is no longer used after #2299 , so maybe we should also remove releated code about `IterType::Gather` or is there some special reason to keep it?

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

**Comment by naoyam** [comment]
Thanks. It should be removed. Can you also remove it from the `IterType` enum?

**Comment by liqiangxl** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

File: csrc/type.cpp [hunk 1]
```diff
@@ -795,8 +795,6 @@ static const char* iter_type2string(IterType t) {
       return "r";
     case IterType::Broadcast:
       return "b";
-    case IterType::Gather:
-      return "g";
     case IterType::Stride:
```
Comment (zasdfgbnm) [inline csrc/type.cpp:798]: Is `IterType::Stride` still used?

Comment (naoyam) [reply csrc/type.cpp:798]: Apparently, no. I think we used to have some conv-like tests by using that, like strided pooling. Stride is fairly simple, so I think we could just leave it there for now, but I'd be ok to remove it as well.

**Pull request merged**